src/Database/Attach/File.php

```diff
@@ -144,6 +145,24 @@ public function fromData($data, $filename)
         return $file;
     }
 
+    /**
+     * Creates a file object from url
+     * @param $url string URL
+     * @return $this
+     */
+    public function fromUrl($url)
```

Comment:
Please also add a `$filename = false` optional parameter that can override the default filename generated from the URL basename. Don't forget to add it to the documentation as well.